### PR TITLE
Fractional brownian motion model

### PR DESCRIPTION
**Pull request by regislebrun**
This PR provides the covariance model of a multivariate fractional Brownian motion when used in a Gaussian process. This model has many interesting properties, eg it allows to model both long range and short range temporal dependence.
It also provides some minor enhancements of the illustrations of the user manual.

### CODE REVIEW

File: lib/src/Base/Stat/FractionalBrownianMotionModel.cxx [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+  const Scalar sOverTheta = s[0] / scale_[0];
+  const Scalar tOverTheta = t[0] / scale_[0];
+  const Scalar stOverTheta = tOverTheta - sOverTheta;
+  const Scalar absSOverTheta = std::abs(sOverTheta);
+  const Scalar absTOverTheta = std::abs(tOverTheta);
+  const Scalar absSTOverTheta = std::abs(stOverTheta);
+  const Scalar sgnSOverTheta = sOverTheta > 0.0 ? 1.0 : -1.0;
+  const Scalar sgnTOverTheta = tOverTheta > 0.0 ? 1.0 : -1.0;
+  const Scalar sgnSTOverTheta = stOverTheta > 0.0 ? 1.0 : -1.0;
+  std::cerr << "sOverTheta=" << sOverTheta << ", tOverTheta=" << tOverTheta << ", stOverTheta=" << stOverTheta << std::endl;
```
Comment (jschueller) [inline lib/src/Base/Stat/FractionalBrownianMotionModel.cxx]: Remains of debug : ) ?

File: lib/src/Base/Stat/FractionalBrownianMotionModel.cxx [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    std::cerr << "i=" << i << ", Hi=" << Hi << ", sigmaI=" << sigmaI << std::endl;
+    for (UnsignedInteger j = 0; j < i; ++j)
+    {
+      const Scalar Hij = Hi + exponent_[j];
+      const Scalar sPowHij = std::pow(absSOverTheta, Hij);
+      const Scalar tPowHij = std::pow(absTOverTheta, Hij);
+      const Scalar stPowHij = std::pow(absSTOverTheta, Hij);
+      const Scalar correlation = rho_(i, j) * (sPowHij + tPowHij - stPowHij);
+      Scalar asymmetry = 0.0;
+      if (std::abs(Hij - 1.0) < 1e-7)
```
Comment (jschueller) [inline lib/src/Base/Stat/FractionalBrownianMotionModel.cxx]: This constant is hardcoded, a comment on this ?

File: lib/src/Base/Stat/FractionalBrownianMotionModel.cxx (hunk 1 above)
Comment (regislebrun) [reply lib/src/Base/Stat/FractionalBrownianMotionModel.cxx]: Oupss sorry. Fixed.

File: lib/src/Base/Stat/FractionalBrownianMotionModel.cxx (hunk 2 above)
Comment (regislebrun) [reply lib/src/Base/Stat/FractionalBrownianMotionModel.cxx]: I added a comment saying that this threshold gives full double precision (according to Maple)

**Comment by jschueller** [comment]
needs a rebase

**Comment by jschueller** [comment]
There should not be a merge commit, please rebase.

**Comment by regislebrun** [comment]
@jschueller I used the github interface to fix the conflict because I was working on another branch, and a merge was the only option. I will fix it if the way to do it is not too steep.

File: python/src/FractionalBrownianMotionModel_doc.i.in [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+eta : 2-d sequence of floats
+    Disymmetry matrix. This matrix express the antisymmetric part of the dependence between the components of the model. It is antisymmetric, only its strictly lower part is addressed.
+rho : 2-d sequence of floats
+    Correlation matrix. This matrix express the symmetric part of the dependence between the components of the model.
+"
+
+// ---------------------------------------------------------------------
+
+%feature("docstring") OT::FractionalBrownianMotionModel::getP
+"P accessor.
```
Comment (sofianehaddad) [inline python/src/FractionalBrownianMotionModel_doc.i.in]: P accessor ? I can not see that method
Btw, `getExponent`, `getRho`, `getEta` are not documented

File: python/test/t_CovarianceModel_std.expout [hunk 4]
```diff
@@ -115,6 +115,26 @@ parameterDescription= [scale_0,scale_1,amplitude_0]
 myModel( [-1,-2] ,  [3,5] )= class=CovarianceMatrix dimension=1 implementation=class=MatrixImplementation name=Unnamed rows=1 columns=1 values=[0.0471566]
 dCov = class=Matrix implementation=class=MatrixImplementation name=Unnamed rows=2 columns=1 values=[0.0490014,0.0857524]
 dCov (FD)= class=Point name=Unnamed dimension=2 values=[0.0490015,0.0857528]
+dCov/dP= [[ 0.098   ]
+ [ 0.3001  ]
+ [ 0.01572 ]]
+myDefautModel =  FractionalBrownianMotionModel(scale=[2], amplitude=[3], exponent=[0.25])
+myModel =  FractionalBrownianMotionModel(scale=[2], amplitude=[3], exponent=[0.25])
+active= [0,1]
+parameter= [2,3]
```
Comment (sofianehaddad) [inline python/test/t_CovarianceModel_std.expout]: Should FractionalBrownianMotionModel::getParameter returns only scale + amplitude ?

Comment (regislebrun) [reply python/test/t_CovarianceModel_std.expout]: done

File: python/src/FractionalBrownianMotionModel_doc.i.in [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+    FractionalBrownianMotionModel(*scale, amplitude, exponent*)
+
+    FractionalBrownianMotionModel(*scale, amplitude, exponent, eta, rho*)
+
+
+Parameters
+----------
+scale : positive float
+    Correlation scale btween two locations.
```
Comment (jschueller) [inline python/src/FractionalBrownianMotionModel_doc.i.in]: btween

File: python/src/FractionalBrownianMotionModel_doc.i.in [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+amplitude : sequence of positive floats
+    Standard deviations of the model :math:`\vect{\sigma}\in \Rset^d`.
+exponent : sequence of float, :math:`0<exponent_i<1`
+    Hurst exponents of the model, ie homogeneity degrees of the self-similarity property.
+eta : :class:`~openturns.SquareMatrix`
+    Disymmetry matrix. This matrix express the antisymmetric part of the dependence between the components of the model. It is antisymmetric, only its strictly lower part is addressed.
+rho : :class:`~openturns.CorrelationMatrix`
+    Correlation matrix. This matrix express the symmetric part of the dependence between the components of the model.
```
Comment (jschueller) [inline python/src/FractionalBrownianMotionModel_doc.i.in]: expresses

File: python/src/FractionalBrownianMotionModel_doc.i.in [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+Parameters
+----------
+scale : positive float
+    Correlation scale btween two locations.
+amplitude : sequence of positive floats
+    Standard deviations of the model :math:`\vect{\sigma}\in \Rset^d`.
+exponent : sequence of float, :math:`0<exponent_i<1`
+    Hurst exponents of the model, ie homogeneity degrees of the self-similarity property.
+eta : :class:`~openturns.SquareMatrix`
+    Disymmetry matrix. This matrix express the antisymmetric part of the dependence between the components of the model. It is antisymmetric, only its strictly lower part is addressed.
```
Comment (jschueller) [inline python/src/FractionalBrownianMotionModel_doc.i.in]: expresses

File: python/src/FractionalBrownianMotionModel_doc.i.in [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      C_{i,j}(s, t)=&\dfrac{\sigma_i\sigma_j}{2}\left\{\rho_{i,j}\left(\left|\dfrac{s}{\theta}\right|^{H_{ij}}+\left|\dfrac{t}{\theta}\right|^{H_{ij}}-\left|\dfrac{t-s}{\theta}\right|^{H_{ij}}\right)-\eta_{i,j}(H_{ij}-1)\left(\sgn(s)\left|\dfrac{s}{\theta}\right|^{H_{ij}}-\sgn(t)\left|\dfrac{t}{\theta}\right|^{H_{ij}}+\sgn(t-s)\left|\dfrac{t-s}{\theta}\right|^{H_{ij}}\right)\right\}
+    \end{align*}
+
+for :math:`H_{ij}=H_i+H_j\neq 1` and:
+
+.. math::
+
+    \forall s, t\in \Rset, C_{i,j}(s, t)=&\dfrac{\sigma_i\sigma_j}{2}\left\{\rho_{i,j}\left(\left|\dfrac{s}{\theta}\right|+\left|\dfrac{t}{\theta}\right|-\left|\dfrac{t-s}{\theta}\right|\right)-\dfrac{\eta_{i,j}}{\theta}\left(s\log|s|-t\log|t|+(t-s)\log|t-s|\right)\right\}
+
+for :math:`H_i + H_j=1`, where :math:`H_i` is the Hurst exponent of the :math:`i`-th component and :math:`\sigma_i` its amplitude. Not that the scale coefficient simplifies in the antisymmetric part when :math:`H_{ij}=1`. The compatibility conditions between the vector of exponents, the correlation matrix and the disymmetry matrix are quite evolved, see [Amblard2012]_.
```
Comment (jschueller) [inline python/src/FractionalBrownianMotionModel_doc.i.in]: try to keep lines <80 chars

**Comment by regislebrun** [comment]
Something else to do here?

**Pull request merged**